src/helpers/GetPreparedPeople.tsx

```diff
@@ -0,0 +1,102 @@
+import { useSearchParams } from 'react-router-dom';
+import { Person } from '../types';
+
+type Sort = 'name' | 'sex' | 'born' | 'died';
+type Sex = 'm' | 'f';
+
+const sortPeople = (
+  people: Person[],
+  sort: Sort,
+  order: boolean,
+) => {
+  switch (sort) {
+    case 'name':
+    case 'sex': {
+      return order
+        ? [...people].sort(
+          (person1, person2) => person2[sort].localeCompare(person1[sort]),
+        )
+        : [...people].sort(
+          (person1, person2) => person1[sort].localeCompare(person2[sort]),
+        );
+    }
+
+    case 'born':
+    case 'died': {
+      return order
+        ? [...people]
+          .sort((person1, person2) => person2[sort] - person1[sort])
+        : [...people]
+          .sort((person1, person2) => person1[sort] - person2[sort]);
+    }
+
+    default: {
+      return [...people];
+    }
+  }
+};
+
+const sexFilter = (
+  people: Person[],
+  sex: Sex,
+) => {
+  return sex === 'm'
+    ? [...people].filter(person => person.sex === 'm')
+    : [...people].filter(person => person.sex === 'f');
+};
+
+const centuriesFilter = (
+  people: Person[],
+  centuries: Array<string>,
+) => {
+  return [...people]
+    .filter(person => centuries.includes(
+      Math.floor(person.born / 100).toString(),
+    ));
+};
+
+const queryFilter = (
+  people: Person[],
+  query: string,
+) => {
+  const lowerCaseQuery = query.toLowerCase();
+
+  return [...people].filter(
+    person => person.name.toLowerCase().includes(lowerCaseQuery)
+    || (person.motherName
+      && person.motherName.toLowerCase().includes(lowerCaseQuery))
+    || (person.fatherName
+      && person.fatherName.toLowerCase().includes(lowerCaseQuery)),
+  );
+};
+
+export const getPreparedPeople = (
+  people: Person[],
+) => {
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const [searchParams] = useSearchParams();
```

Comment:
you must dont use useSearchParams  here, but just pass attribute (sort, order, etc) in func